src/cargo/util/toml.rs

```diff
@@ -540,6 +540,18 @@ impl TomlManifest {
             }
         }
 
+        {
+            let mut names_sources = HashMap::new();
+            for dep in deps.iter() {
+                let name = dep.name();
+                let sources = names_sources.entry(name).or_insert(HashSet::new());
+                if sources.insert(dep.source_id()) && sources.len() > 1 {
```

Comment:
Yes, it is. But this line means "if insertion was successful, and we now has more than 1 source". So we check sources length only if we inserted something. Actually, I thought to split this line on two because we mostly care about number of sources (second part of the condition).

We can miss different sources if we will check it with `!sources.insert(source_id)`. Say, we have source1 and source2 for the same package. On both iterations it will return `false` and if-branch will not be executed.

So, would it be better to leave it as is or split it on two lines?